packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -220,6 +228,7 @@ class FlutterCommandRunner extends CommandRunner {
     // Set up the tooling configuration.
     if (enginePath != null) {
       ToolConfiguration.instance.engineSrcPath = enginePath;
+      ToolConfiguration.instance.engineBuildPath = _findEngineBuildPath(globalResults, enginePath);
 
       if (globalResults.wasParsed('engine-release'))
         ToolConfiguration.instance.engineRelease = globalResults['engine-release'];
```

Comment:
This is still used to find target-specific tools
(A future PR will use --engine-build to locate these and will remove --engine-debug/--engine-release)